node/src/p2p/p2p_effects.rs

```diff
@@ -231,28 +229,22 @@ pub fn node_p2p_effects<S: Service>(store: &mut Store<S>, action: P2pActionWithM
                     _ => {}
                 }
             }
-            P2pChannelsAction::Snark(action) => {
-                // TODO: does the order matter here? if not this clone can be removed
-                action.clone().effects(&meta, store);
-                match action {
-                    P2pChannelsSnarkAction::Received { peer_id, snark } => {
-                        store.dispatch(SnarkPoolCandidateAction::InfoReceived {
-                            peer_id,
-                            info: *snark,
-                        });
-                    }
-                    P2pChannelsSnarkAction::Libp2pReceived { peer_id, snark, .. } => {
-                        store.dispatch(SnarkPoolCandidateAction::WorkReceived {
-                            peer_id,
-                            work: *snark,
-                        });
-                    }
-                    _ => {}
+            P2pChannelsAction::Snark(action) => match action {
+                P2pChannelsSnarkAction::Received { peer_id, snark } => {
+                    store.dispatch(SnarkPoolCandidateAction::InfoReceived {
+                        peer_id,
+                        info: *snark,
+                    });
                 }
-            }
+                P2pChannelsSnarkAction::Libp2pReceived { peer_id, snark, .. } => {
+                    store.dispatch(SnarkPoolCandidateAction::WorkReceived {
+                        peer_id,
+                        work: *snark,
+                    });
+                }
+                _ => {}
+            },
             P2pChannelsAction::SnarkJobCommitment(action) => {
-                // TODO: does the order matter here? if not this clone can be removed
```

Comment:
@0xMimir I guess I will figure out by reading the rest of the diff, but I was wondering where did these `effects` calls go

Reply:
Effects that don't require service are in reducer and ones that do are in `P2pChannelsSnarkJobCommitmentEffectfulAction::effects`.